lib/model/query/datasets.js

```diff
@@ -153,41 +179,63 @@ createPublishedProperty.audit = (property, dataset) => (log) =>
-WITH properties_update as (
-  UPDATE ds_properties dp SET "publishedAt" = ${publishedAt}
-  FROM ds_property_fields dpf
-  WHERE dpf."formDefId" = ${formDefId}
-  AND dpf."dsPropertyId" = dp.id
-  AND dp."publishedAt" IS NULL
-  RETURNING dp.*
-), datasets_update as (
-  UPDATE datasets ds SET "publishedAt" = ${publishedAt}
-  FROM dataset_form_defs dfd
+const publishIfExists = (formDefId, publishedAt) => async ({ all, maybeOne }) => {
+
+  // Helper sub-queries
+  const _publishIfExists = () => sql`
+  WITH properties_update as (
+    UPDATE ds_properties dp SET "publishedAt" = ${publishedAt}
+    FROM ds_property_fields dpf
+    WHERE dpf."formDefId" = ${formDefId}
+    AND dpf."dsPropertyId" = dp.id
+    AND dp."publishedAt" IS NULL
+    RETURNING dp.*
+  ), datasets_update as (
+    UPDATE datasets ds SET "publishedAt" = ${publishedAt}
+    FROM dataset_form_defs dfd
+    WHERE dfd."formDefId" = ${formDefId}
+    AND dfd."datasetId" = ds.id
+    AND ds."publishedAt" IS NULL
+    RETURNING *
+  )
+  -- selecting following for publish.audit
+  -- we are returning first row for dataset
+  -- because we want to log it even if no
+  -- property is published.
+  SELECT -1 "propertyId", '' "name", ds.id, ds."acteeId", 'datasetCreated' action
+  FROM datasets_update ds
+  UNION
+  SELECT p.id "propertyId", p.name, ds.id, ds."acteeId", 'propertyAdded' action
+  FROM properties_update p
+  JOIN datasets ds on ds.id = p."datasetId"
+  UNION
+  SELECT p.id "propertyId", p.name, ds.id, ds."acteeId", 'noop' action
+  FROM ds_properties p
+  JOIN datasets ds on ds.id = p."datasetId"
+  JOIN dataset_form_defs dfd on dfd."datasetId" = ds.id
   WHERE dfd."formDefId" = ${formDefId}
-  AND dfd."datasetId" = ds.id
-  AND ds."publishedAt" IS NULL
-  RETURNING *
-)
--- selecting following for publish.audit
--- we are returning first row for dataset
--- because we want to log it even if no
--- property is published.
-SELECT -1 "propertyId", '' "name", ds.id, ds."acteeId", 'datasetCreated' action
-FROM datasets_update ds
-UNION
-SELECT p.id "propertyId", p.name, ds.id, ds."acteeId", 'propertyAdded' action
-FROM properties_update p
-JOIN datasets ds on ds.id = p."datasetId"
-UNION
-SELECT p.id "propertyId", p.name, ds.id, ds."acteeId", 'noop' action
-FROM ds_properties p
-JOIN datasets ds on ds.id = p."datasetId"
-JOIN dataset_form_defs dfd on dfd."datasetId" = ds.id
-WHERE dfd."formDefId" = ${formDefId}
-AND p."publishedAt" IS NOT NULL
-ORDER BY "propertyId"
-`);
+  AND p."publishedAt" IS NOT NULL
+  ORDER BY "propertyId"
+  `;
+
+  const _datasetNameConflict = () => sql`
+    SELECT existing_ds.name as current, ds_to_publish.name as provided FROM dataset_form_defs dfd
+    JOIN datasets ds_to_publish ON dfd."datasetId" = ds_to_publish.id
+    JOIN datasets existing_ds ON LOWER(ds_to_publish.name) = LOWER(existing_ds.name)
+      AND ds_to_publish.id != existing_ds.id
+      AND existing_ds."publishedAt" IS NOT NULL
+      AND ds_to_publish.name != existing_ds.name
```

Comment:
Condition not necessary, dataset already exists at this point so it wont be possible for two with the same name to already exist.